compiler/luci-interpreter/src/kernels/Quantize.cpp

```diff
@@ -0,0 +1,230 @@
+  {
+    LUCI_INTERPRETER_CHECK(output()->element_type() == loco::DataType::U8 ||
+                           output()->element_type() == loco::DataType::S8 ||
+                           output()->element_type() == loco::DataType::S16);
+  }
+  else if (input()->element_type() == loco::DataType::S16)
+  {
+    LUCI_INTERPRETER_CHECK(output()->element_type() == loco::DataType::S8 ||
+                           output()->element_type() == loco::DataType::S16 ||
+                           output()->element_type() == loco::DataType::S32);
+    if (output()->element_type() == loco::DataType::S16)
+    {
+      LUCI_INTERPRETER_CHECK(input()->zero_point() == 0 && output()->zero_point() == 0);
+    }
+  }
+  else if (input()->element_type() == loco::DataType::S8 ||
+           input()->element_type() == loco::DataType::U8)
+  {
+    LUCI_INTERPRETER_CHECK(output()->element_type() == loco::DataType::S8 ||
+                           output()->element_type() == loco::DataType::U8);
+  }
+  else
+  {
+    throw std::runtime_error("Unsupported type.");
+  }
+
+  output()->resize(input()->shape());
+}
+
+void Quantize::execute() const
+{
+  switch (input()->element_type())
+  {
+    case loco::DataType::FLOAT32:
+    {
+      tflite::QuantizationParams op_params;
+      op_params.zero_point = output()->zero_point();
+      op_params.scale = output()->scale();
+      const auto input_data = getTensorData<float>(input());
+
+      switch (output()->element_type())
+      {
+        case loco::DataType::S8:
+        {
+          luci_interpreter_pal::Quantize(op_params, getTensorShape(input()), input_data,
+                                         getTensorShape(output()), getTensorData<int8_t>(output()));
+          break;
+        }
+        case loco::DataType::U8:
+        {
+          luci_interpreter_pal::Quantize(op_params, getTensorShape(input()), input_data,
+                                         getTensorShape(output()),
+                                         getTensorData<uint8_t>(output()));
+          break;
+        }
+        case loco::DataType::S16:
+        {
+          luci_interpreter_pal::Quantize(op_params, getTensorShape(input()), input_data,
+                                         getTensorShape(output()),
+                                         getTensorData<int16_t>(output()));
+          break;
+        }
+        default:
+          throw std::runtime_error("Unsupported type.");
+      }
+      break;
+    }
+    case loco::DataType::S16:
+    {
+      int32_t multiplier;
+      int32_t shift;
+
+      compute_shift_multiplier(static_cast<double>(input()->scale()),
+                               static_cast<double>(output()->scale()), &multiplier, &shift);
+
+      const auto input_shape = getTensorShape(input());
+      const auto output_shape = getTensorShape(output());
+
+      const auto input_data = getTensorData<int16_t>(input());
+
+      const auto size = tflite::MatchingFlatSize(input_shape, output_shape);
```

Comment:
I think we could extract this lines (except getTensorData<int16>) from swith-case in order to reduce code duplication.